### PR TITLE
Ui optimization

### PR DESCRIPTION
**Pull request by Rushi98**
- mostly simplified the layouts
- moved sort button in course list fragment to action bar
- course viewholder : card heavy --> simpler UI

### CODE REVIEW

**Comment by Rushi98** [comment]
part of #67

**Comment by aayushsingla** [comment]
Your changes caused this crash last night. Please do changes keeping this in mind.
`Fatal Exception: java.lang.ClassCastException: com.macbitsgoa.comrades.homefragment.HomeFragment cannot be cast to com.macbitsgoa.comrades.courselistfragment.CourseListFragment
       at com.macbitsgoa.comrades.HomeActivity$2.onMenuItemClick(HomeActivity.java:226)
       at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:154)
       at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:991)
       at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:981)
       at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:625)
       at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:151)
       at android.view.View.performClick(View.java:5640)
       at android.view.View$PerformClick.run(View.java:22455)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6165)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:888)
       at `com.android.internal.os.ZygoteInit.main(ZygoteInit.java:778)``

File: androidClient/app/src/main/java/com/macbitsgoa/comrades/HomeActivity.java [hunk 1]
```diff
@@ -36,59 +37,71 @@
...
                 fab_add_course.setVisibility(View.GONE);
                 fab_add_course.setOnClickListener(null);
-                Fragment homeFragment=fragmentManager.findFragmentByTag("HomeFragment");
-                if(homeFragment==null){
-                    homeFragment=HomeFragment.newInstance();
+                Fragment homeFragment = fragmentManager.findFragmentByTag(TAG_HOME_FRAG);
+                if (homeFragment == null) {
+                    homeFragment = HomeFragment.newInstance();
                 }
                 fragmentManager.beginTransaction().replace(R.id.container_fragment,
```
Comment (aayushsingla) [inline androidClient/app/src/main/java/com/macbitsgoa/comrades/HomeActivity.java]: Please call `addToBackstack(null)` here

**Comment by aayushsingla** [comment]
Also, due to sort button in toolbar, Search bar doesnot expand to its maximum width.

**Comment by Rushi98** [comment]
@aayushsingla does the exception still happen?

**Pull request merged**